### PR TITLE
Add auth isolation for oauth flows

### PR DESCRIPTION
**Pull request by samhorlbeck**

### CODE REVIEW

File: packages-exp/auth-exp/src/core/strategies/redirect.ts [hunk 1]
```diff
@@ -172,3 +172,7 @@ class RedirectAction extends AbstractPopupRedirectOperation {
 
   cleanUp(): void {}
 }
+
+export function _clearOutcomes(): void {
+  redirectOutcomeMap = {};
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/core/strategies/redirect.ts:177]: Can you call https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/clear instead and keep this is a const?

File: packages-exp/auth-exp/src/platform_browser/popup_redirect.ts [hunk 2]
```diff
@@ -57,8 +61,11 @@ class BrowserPopupRedirectResolver implements PopupRedirectResolver {
     authType: AuthEventType,
     eventId?: string
   ): Promise<AuthPopup> {
+    console.log(this.eventManagers);
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/platform_browser/popup_redirect.ts:64]: stray logs

Comment (samhorlbeck) [reply packages-exp/auth-exp/src/platform_browser/popup_redirect.ts]: Done

File: packages-exp/auth-exp/src/core/strategies/redirect.ts (hunk 1 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/strategies/redirect.ts]: Done

**Pull request merged**